src/pineko/comparator.py

```diff
@@ -32,19 +35,68 @@ def compare(pine, fktable, max_as, max_al, pdf, xir, xif):
     """
     import lhapdf  # pylint: disable=import-error,import-outside-toplevel
 
-    pdfset = lhapdf.mkPDF(pdf, 0)
-    pdgid = int(pdfset.set().get_entry("Particle"))
+    pdfset1 = lhapdf.mkPDF(pdf1, 0)
+    if pdf2 is not None:
+        pdfset2 = lhapdf.mkPDF(pdf2, 0)
+    else:
+        pdfset2 = pdfset1
+
+    # TODO: This should probably changed in the future to use the Grid::convolutions
+    try:
+        parton1 = int(pine.key_values()["convolution_particle_1"])
+        parton2 = int(pine.key_values()["convolution_particle_2"])
+
+        if (
+            fktable.key_values()["convolution_type_1"]
+            != pine.key_values()["convolution_type_1"]
+            or fktable.key_values()["convolution_type_2"]
+            != pine.key_values()["convolution_type_2"]
+        ):
+            raise ValueError(
+                "Grids and FkTables do not have the same convolution types"
+            )
+
+        # log some useful info to check if PDFs are swapped
+        rich.print(
+            f"[yellow]Convolution type 1: {fktable.key_values()['convolution_type_1']}, PDF 1: {pdf1}"
+        )
+        rich.print(
+            f"[yellow]Convolution type 2: {fktable.key_values()['convolution_type_2']}, PDF 2: {pdf2}"
+        )
+
+    except KeyError:
+        parton1 = int(pine.key_values()["initial_state_1"])
+        parton2 = int(pine.key_values()["initial_state_2"])
+    hadronic = parton1 == parton2
+
     order_mask = pineappl.grid.Order.create_mask(pine.orders(), max_as, max_al, True)
-    before = np.array(
-        pine.convolute_with_one(
-            pdgid,
-            pdfset.xfxQ2,
-            pdfset.alphasQ2,
-            order_mask=order_mask,
-            xi=((xir, xif),),
+    if hadronic:
+        before = np.array(
+            pine.convolve_with_two(
+                pdg_id1=parton1,
+                xfx1=pdfset1.xfxQ2,
+                pdg_id2=parton2,
+                xfx2=pdfset2.xfxQ2,
+                alphas=pdfset1.alphasQ2,
```

Comment:
I made this choice, I mean `alpha_s` evolution does not depend on polarized or timelike. 
I guess you need to chose one of the 2 pdfs or do you really want to get it from eko?

Reply:
No, wanted just to know, since in principle I can be comparing with two PDFs that have non-compatible values of alpha.

Reply:
> No, wanted just to know, since in principle I can be comparing with two PDFs that have non-compatible values of alpha.

this could happen, but since it is hidden behind LHAPDF it is hard to find out (the only way would be to compare the whole function)

Reply:
So it is mostly hoping that it is compatible. But ok, this is just the comparison so hopefully upon discovering there's a problem this is one thing people would look into... maybe an info print or something like that would be good in this case.